File: frontend/app/lib/api/numbering.ts
import { getJSON, postJSON, patchJSON } from "./submissions";
import { APIResult } from "../types/data";
import { NumberingScheme, CreateNumberingSchemeRequest, UpdateNumberingSchemeRequest } from "../types/numbering";

export async function getSchemes(): Promise<APIResult<NumberingScheme[]>> {
	return getJSON("/numbering");
}

export async function createScheme(body: CreateNumberingSchemeRequest): Promise<APIResult<NumberingScheme>> {
	return postJSON("/numbering", body);
}

export async function updateScheme(id: number, body: UpdateNumberingSchemeRequest): Promise<APIResult<NumberingScheme>> {
	return patchJSON(`/numbering/${id}`, body);
}

export async function previewNext(documentType: string): Promise<APIResult<{ next: string }>> {
	return getJSON(`/numbering/preview/${documentType}`);
}

File: backend/app/routers/numbering.py
from fastapi import APIRouter, Depends
from app.services import numbering
from app.services.session import require_permission
from app.types.numbering import CreateNumberingSchemeRequest, UpdateNumberingSchemeRequest

router = APIRouter()


@router.get("")
async def list_schemes(_=Depends(require_permission("System.Read"))):
    return await numbering.list_schemes()


@router.post("")
async def create_scheme(data: CreateNumberingSchemeRequest, _=Depends(require_permission("System.Write"))):
    return await numbering.create_scheme(data)


@router.patch("/{scheme_id}")
async def update_scheme(scheme_id: int, data: UpdateNumberingSchemeRequest, _=Depends(require_permission("System.Write"))):
    return await numbering.update_scheme(scheme_id, data)


@router.get("/preview/{document_type}")
async def get_preview(document_type: str, _=Depends(require_permission("System.Read"))):
    return await numbering.get_preview(document_type)
